integration/csv_fork_test.go

```diff
@@ -0,0 +1,687 @@
+	b := txscript.NewScriptBuilder()
+	b.AddInt64(int64(sequenceLock))
+	b.AddOp(txscript.OP_CHECKSEQUENCEVERIFY)
+	b.AddOp(txscript.OP_DROP)
+	csvScript, err := b.Script()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Using the script generated above, create a P2SH output which will be
+	// accepted into the mempool.
+	p2shAddr, err := btcutil.NewAddressScriptHash(csvScript, r.ActiveNet)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	p2shScript, err := txscript.PayToAddrScript(p2shAddr)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	output := &wire.TxOut{
+		PkScript: p2shScript,
+		Value:    int64(numSatoshis),
+	}
+
+	// Finally create a valid transactions which creates the output crafted
+	// above.
+	tx, err := r.CreateTransaction([]*wire.TxOut{output}, 10)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var outputIndex uint32
+	if bytes.Equal(tx.TxOut[0].PkScript, p2shScript) {
+		outputIndex = 0
+	} else {
+		outputIndex = 1
+	}
+
+	utxo := &wire.OutPoint{
+		Hash:  tx.TxHash(),
+		Index: outputIndex,
+	}
+
+	return csvScript, utxo, tx, nil
+}
+
+// spendCSVOutput spends an output previously created by the createCSVOutput
+// function. The sigScript is a trivial push of OP_TRUE followed by the
+// redeemScript to pass P2SH evaluation.
+func spendCSVOutput(redeemScript []byte, csvUTXO *wire.OutPoint,
+	sequence uint32, targetOutput *wire.TxOut,
+	txVersion int32) (*wire.MsgTx, error) {
+
+	tx := wire.NewMsgTx(txVersion)
+	tx.AddTxIn(&wire.TxIn{
+		PreviousOutPoint: *csvUTXO,
+		Sequence:         sequence,
+	})
+	tx.AddTxOut(targetOutput)
+
+	b := txscript.NewScriptBuilder()
+	b.AddOp(txscript.OP_TRUE)
+	b.AddData(redeemScript)
+
+	sigScript, err := b.Script()
+	if err != nil {
+		return nil, err
+	}
+	tx.TxIn[0].SignatureScript = sigScript
+
+	return tx, nil
+}
+
+// isSoftForkActive returns true if the soft-fork as defined by the passed
+// string is currently active.
+func isSoftForkActive(r *rpctest.Harness, t *testing.T, bip9ForkName string) bool {
+	chainInfo, err := r.Node.GetBlockChainInfo()
+	if err != nil {
+		t.Fatalf("unable to query for chain info: %v", err)
+	}
+
+	csvForkState, ok := chainInfo.Bip9SoftForks[bip9ForkName]
+	if !ok {
+		t.Fatalf("%v soft-fork state not found in getblockchaininfo "+
+			"response", bip9ForkName)
+	}
+
+	return csvForkState.Status == "active"
+}
+
+// assertTxInBlock asserts a transaction with the specified txid is found
+// within the block with the passed block hash.
+func assertTxInBlock(r *rpctest.Harness, t *testing.T, blockHash *chainhash.Hash,
+	txid *chainhash.Hash) {
+
+	block, err := r.Node.GetBlock(blockHash)
+	if err != nil {
+		t.Fatalf("unable to get block: %v", err)
+	}
+	if len(block.Transactions) < 2 {
+		t.Fatal("target transaction was not mined")
+	}
+
+	for _, txn := range block.Transactions {
+		txHash := txn.TxHash()
+		if (&txHash).IsEqual(txid) {
+			return
+		}
+	}
+
+	t.Fatalf("txid %v was not found in block %v", txid, blockHash)
```

Comment:
Fixed, also added similar caller/line context to the `isSoftForkActive` function.